File: app/src/main/java/com/example/canis/BeaconModule/BeaconNavigationModule/InstanceProvider.java
package com.example.canis.BeaconModule.BeaconNavigationModule;

import com.example.canis.BeaconModule.BeaconNavigationModule.network.BeaconNavigationController;
import com.example.canis.BeaconModule.BeaconNavigationModule.service.BeaconNavigationService;
import com.example.canis.RestServiceBuilder;
import com.example.canis.RestServiceBuilder2;

public class InstanceProvider {

    public static BeaconNavigationService getBuildingMapService() {
        return new BeaconNavigationService(RestServiceBuilder2.INSTANCE.build(BeaconNavigationController.class));
    }
}

File: app/src/main/java/com/example/canis/WorkersInformationModule/InstanceProvider.kt
package com.example.canis.WorkersInformationModule

import com.example.canis.RestServiceBuilder
import com.example.canis.WorkersInformationModule.data.RestWorkerProvider
import com.example.canis.WorkersInformationModule.data.WorkersDataProvider
import com.example.canis.WorkersInformationModule.model.WorkersService
import com.example.canis.WorkersInformationModule.network.WorkersInformationService
import com.example.canis.WorkersInformationModule.network.WorkersListService
import org.koin.dsl.module

object InstanceProvider {

    val commonModule = module {

        single { RestServiceBuilder.build(WorkersInformationService::class.java) }
        single { RestServiceBuilder.build(WorkersListService::class.java)}
        single<WorkersDataProvider> {RestWorkerProvider(get(), get())}
        single { WorkersService(get())}

    }
}